### PR TITLE
Sapling verifying keys

### PR DESCRIPTION
**Pull request by svyatonik**
This PR introduces:
- sapling (spend, output) verifying keys in JSON format;
- tests for sapling verification.

### CODE REVIEW

File: chain/src/block.rs [hunk 1]
```diff
@@ -68,6 +68,13 @@ mod tests {
...
 				"000000e869e3a0fa79858a51b4b1d09a6480dcdb37bae63653fcb11a718abf3f",
 				"6476c40e8de0771a76db44e1d9b3f0a79bd0307e0174245af5b8923a1142f712",
+			),
+			(
+				// https://zcash.blockexplorer.com/block/00000000007ef95f986ed8309d0ed6a1b6174c90b9c7f4d0dfc40f7147315e79
```
Comment (svyatonik) [inline chain/src/block.rs]: had problems retrieving tx using `rawtx` API => added this test to make sure we deserialize/serialize block correctly

File: crypto/Cargo.toml [hunk 2]
```diff
@@ -4,14 +4,15 @@ version = "0.1.0"
 authors = ["debris <marek.kotewicz@gmail.com>"]
 
 [dependencies]
-blake2-rfc = { git = "https://github.com/gtank/blake2-rfc.git", branch = "persona" }
+bellman = "0.1"
```
Comment (svyatonik) [inline crypto/Cargo.toml]: thought it is better to move all crypto-specific dependencies from `verification` here

File: message/src/common/inventory.rs [hunk 3]
```diff
@@ -115,10 +115,10 @@ mod tests {
 
 	#[test]
 	fn test_inventory_type_conversion() {
-		assert_eq!(0u32, InventoryType::Error.into());
-		assert_eq!(1u32, InventoryType::MessageTx.into());
-		assert_eq!(2u32, InventoryType::MessageBlock.into());
-		assert_eq!(3u32, InventoryType::MessageFilteredBlock.into());
+		assert_eq!(0u32, u32::from(InventoryType::Error));
```
Comment (svyatonik) [inline message/src/common/inventory.rs]: some magic - this change is completely unrelated to all other changes, but I'm getting compilation error:
```
error[E0282]: type annotations needed                                                                                                                                                                          
   --> message/src/common/inventory.rs:118:20                                                                                                                                                                  
    |                                                                                                                                                                                                          
118 |         assert_eq!(0u32, InventoryType::Error.into());                                                                                                                                                   
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for `T` 
```
after this PR (and no compilation errors before)

File: res/sapling-output-verifying-key.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    "alphaG1": "0x0db882cf5db3e8567f16b4db1772d4d1f5a3fe8d62f0df2eb8a5cfa50806702afde8fc25335eb5ec859c2818b2610b2e19ab445dac720bb1f2b0cd3336f7a1acc62bf1b3a321826264dc7e469281e23b218394d598689da04e136878ff9a7897",
```
Comment (svyatonik) [inline res/sapling-output-verifying-key.json]: `x` and `y` are merged here into single byte array - haven't found an API to construct points from separate `x` and `y`

**Comment by NikVolf** [comment]
`pghr13_json.rs` is rather an inconvenient file name

let's make `json` directory and `pghr13.rs` there?

or just put everything in json.rs combined?

File: res/sapling-output-verifying-key.json (hunk 4 above)
Comment (NikVolf) [reply res/sapling-output-verifying-key.json]: Yeah I had to alter `bn` back in 2017 library to do such API

**Pull request merged**